src/pages/event/EventDetailPage/EventDetailPage.tsx

```diff
@@ -0,0 +1,93 @@
+import getEventDetail from '@/apis/event/getEventDetail';
+
+import { FaRegBookmark } from 'react-icons/fa';
+import { useParams } from 'react-router-dom';
+
+import { useQuery } from '@tanstack/react-query';
+
+import {
+  ButtonWrapper,
+  ContentLabel,
+  DetailContentWrapper,
+  EventDetailWrapper,
+  EventTitle,
+  FormButtonWrapper,
+  PosterImage,
+  PurpleButton,
+  SemiPurpleButton,
+  TwoContentWrapper,
+  UpdateDeleteWrapper,
+} from './EventDetailPage.style';
+
+const EventDetailPage = () => {
+  const { eventId } = useParams();
+  const { data: eventDetail } = useQuery(['event_detail', 'eventId'], () =>
+    getEventDetail({ id: eventId }),
+  );
+
+  if (!eventDetail) {
+    return <div>Loading...</div>;
+  }
+
+  const {
+    title,
+    poster,
+    startDate,
+    startTime,
+    location,
+    openDate,
+    openTime,
+    closeDate,
+    closeTime,
+    name,
+  } = eventDetail;
```

Comment:
```suggestion
  if (!eventDetail) {
    return <div>Loading...</div>;
  }

  const {
    title,
    poster,
    startDate,
    startTime,
    location,
    openDate,
    openTime,
    closeDate,
    closeTime,
    name,
  } = eventDetail!;
```
요렇게 하면 됩니다!
아니면 isLoading을 사용할 수도 있고 커스텀 훅으로 제작해서 defaultValue를 지정해 줄수도 있습니다. 그리고 이 api는 staleTime도 지정해서 캐싱하면 좋을것 같아요! 자주 바뀌는 정보가 아니니까??

그래서 그런 작업들을 위해 저는 query를 사용할때 커스텀 훅으로 제작합니다. 컴포넌트 단에서 분리하여 데이터에 대한 로직을 캡슐화하는거죠! 커스텀 훅으로 바꾸는 방법도 생각해보세요!
https://www.youtube.com/watch?v=MArE6Hy371c
배민에서도 이런 방법을 채택한것 같고, 여러모로 유용한 방법이라고 생각합니다~!

Reply:
감사합니다:) 꼭 한번 시청해봐야겠네요!!